### PR TITLE
Fixes #34197 - reuse find_installed_package instead of rpm_version

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

**Comment by upadhyeammit** [comment]
While including apt support I felt `rpm_version` method can be refactored to use `find_installed_package` . Once apt is available package_manager can selectively query respective `find_installed_package` method as per the OS.

File: lib/foreman_maintain/package_manager/yum.rb [hunk 1]
```diff
@@ -38,8 +38,12 @@ def installed?(packages)
       sys.execute?(%(rpm -q #{packages_list}))
     end
 
-    def find_installed_package(name)
-      status, result = sys.execute_with_status(%(rpm -q '#{name}'))
+    def find_installed_package(name, queryfm = '')
```
Comment (evgeni) [inline lib/foreman_maintain/package_manager/yum.rb:41]: I'd call this param `queryfmt` or `queryformat`, `fmt` is a common abbreviation for "format" and is also used in `man rpm` to describe the *value* passed after `--queryformat`. `fm` on the other hand can mean `foreman-maintain` and that initially confused me.

**Review by evgeni — COMMENTED** [review]
small naming nit, otherwise I like it!

File: lib/foreman_maintain/package_manager/yum.rb (hunk 1 above)
Comment (upadhyeammit) [reply lib/foreman_maintain/package_manager/yum.rb]: Thanks for the review, as I am at changing param name, I changed it for list_installed_packages as well.

File: lib/foreman_maintain/package_manager/yum.rb [hunk 2]
```diff
@@ -76,12 +80,12 @@ def files_not_owned_by_package(directory)
       sys.execute(find_cmd).split("\n")
     end
 
-    def list_installed_packages(queryfm = '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n')
-      # The queryfm should only include valid tag(s) as per `rpm --querytag` list.
+    def list_installed_packages(queryformat = '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n')
+      # The queryformat should only include valid tag(s) as per `rpm --querytag` list.
       # If any special formatting is required with querytag then it should be provided with tag i.e,
       # querytag = "--%{VENDOR}"
```
Comment (evgeni) [inline lib/foreman_maintain/package_manager/yum.rb:86]: what's `querytag` here?

**Review by evgeni — APPROVED** [review]

File: lib/foreman_maintain/package_manager/yum.rb (hunk 2 above)
Comment (upadhyeammit) [reply lib/foreman_maintain/package_manager/yum.rb]: Thanks for the catch, corrected it

File: test/test_helper.rb [hunk 3]
```diff
@@ -63,7 +63,7 @@ def installed?(packages)
     [packages].flatten(1).all? { |p| @packages.include?(p) }
   end
 
-  def find_installed_package(name)
+  def find_installed_package(name, _queryfm = '')
```
[the remark this answers is not in the document]
Comment (upadhyeammit) [reply test/test_helper.rb]: Agree, done!

**Review by evgeni — APPROVED** [review]

**Comment by upadhyeammit** [comment]
@evgeni thanks for review! Merging now!

**Pull request merged**